src/libraries/System.Private.CoreLib/src/System/Globalization/GlobalizationMode.cs

```diff
@@ -13,6 +13,7 @@ private static partial class Settings
         {
             internal static readonly bool PredefinedCulturesOnly = AppContextConfigHelper.GetBooleanConfig("System.Globalization.PredefinedCulturesOnly", "DOTNET_SYSTEM_GLOBALIZATION_PREDEFINED_CULTURES_ONLY");
             internal static bool Invariant { get; } = GetInvariantSwitchValue();
+            internal static readonly bool AllowInvariantCultureOnly = AppContextConfigHelper.GetBooleanConfig("System.Globalization.AllowInvariantCultureOnly", "DOTNET_SYSTEM_GLOBALIZATION_INVARIANT_CULTURE_ONLY");
```

Comment:
It looks like the trimming tests are failing because this new addition is causing the `Settings` class to be preserved when trimming with `Invariant=true`. This will also cause `AppContextConfigHelper` to be preserved in that mode.

@marek-safar - are we certain this new setting will provide enough value to justify adding it?

Reply:
That was my point I raised in the comment https://github.com/dotnet/runtime/issues/43774#issuecomment-715627250. @marek-safar can advise if he still seeing a good value to have this or just not worth it.

Reply:
CC @jkotas for awareness

Reply:
I don't see AllowInvariantCultureOnly setting to be ever used so it'd be better to have it always trimmed.

Reply:
If you don't see it ever being used, why are we adding it?

Reply:
@marek-safar please advise if this feature you originally requested is needed and will be consumed by some platforms/containers. If you think it is not clear where is going to be used, then I would hold on it till it will be clear it is really needed.

Reply:
I think the setting would make more sense as opt-out mode. Right now I don't see we will advise developers which invariant mode they should be using especially if this one is not fully integrated (no msbuild property).

Reply:
I agree with @marek-safar . I think this should be enabled by default for invariant mode. I think it is better to throw exceptions than to return bogus results (current behavior). If somebody is happy with the bogus results, they can opt-in into the somewhat broken behavior.

I understand that it means that some libraries will be broken by default in invariant mode. I think that it is a good thing. It will contribute towards being able to use the invariant mode with confidence.

Reply:
Are we concerned that this can break existing apps using Invariant mode? For example apps running on alpine?

cc @richlander @MichaelSimons

Reply:
Yes, the change would have to be tracked as a .NET 6 breaking change. The apps that got broken and are happy with the bogus globalization behavior can opt-in into it.

Reply:
I am not agreeing with that. what benefit we are getting here more than creating burden on us asking every app running into this problem to enable this switch? why returning invariant data for any locale is bad or wrong?

Reply:
@jkotas @marek-safar my last commit implemented throwing by default when creating any culture different than Invariant culture when enabling the Globalization Invariant Mode. also, I am going to create a breaking change issue and link it to this PR too. Could you please have a look?

As @jkotas suggested we'll use the PredefinedCulturesOnly config switch to control the behavior. I'll add more details in the PR description about that.